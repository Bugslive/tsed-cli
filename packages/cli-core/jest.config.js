// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html

module.exports = {
  ...require("@tsed/jest-config"),
  roots: ["<rootDir>/src", "<rootDir>/test"],
  moduleNameMapper: {
    "@tsed/cli-core": "<rootDir>/src/index.ts"
  },
  coverageThreshold: {
    global: {
      statements: 71.23,
      branches: 72.81,
      functions: 48.9,
      lines: 71.23
    }
  },
};
