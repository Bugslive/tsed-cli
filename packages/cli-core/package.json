{
  "name": "@tsed/cli-core",
  "description": "Build your CLI with TypeScript and Decorators",
  "version": "6.0.0-alpha.8",
  "type": "module",
  "main": "./lib/esm/index.js",
  "source": "./src/index.ts",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "default": "./lib/esm/index.js"
    }
  },
  "scripts": {
    "build": "yarn build:ts",
    "build:ts": "tsc --build tsconfig.json",
    "test": "vitest run",
    "test:ci": "vitest run --coverage.thresholds.autoUpdate=true"
  },
  "keywords": [
    "cli",
    "typescript",
    "framework",
    "api",
    "decorators",
    "commander.js",
    "inquirer.js",
    "listr"
  ],
  "dependencies": {
    "@npmcli/run-script": "8.1.0",
    "@tsed/logger": ">=6.1.1",
    "@tsed/normalize-path": ">=7.14.2",
    "@types/fs-extra": "^11.0.4",
    "@types/inquirer": "^9.0.7",
    "ajv": "8.17.1",
    "axios": "1.6.7",
    "chalk": "4.1.2",
    "change-case": "^4.1.2",
    "commander": "9.3.0",
    "consolidate": "1.0.1",
    "create-frame": "1.0.0",
    "enquirer": "2.3.6",
    "execa": "5.1.1",
    "figures": "4.0.1",
    "fs-extra": "11.1.0",
    "globby": "11.1.0",
    "handlebars": "4.7.7",
    "handlebars-utils": "1.0.6",
    "inquirer": "8.2.4",
    "inquirer-autocomplete-prompt": "2.0.0",
    "js-yaml": "4.1.0",
    "listr2": "5.0.5",
    "normalize-path": "3.0.0",
    "read-pkg-up": "7.0.1",
    "registry-url": "5.1.0",
    "rxjs": "^7.5.5",
    "semver": "7.3.5",
    "split": "1.0.1",
    "tslib": "2.3.1",
    "tunnel": "0.0.6",
    "update-notifier": "7.3.1",
    "url-parse": "1.5.10",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "@tsed/typescript": "workspace:*",
    "@types/axios": "0.14.0",
    "@types/commander": "2.12.2",
    "@types/consolidate": "0.14.4",
    "@types/figures": "3.0.1",
    "@types/globby": "9.1.0",
    "@types/inquirer": "9.0.7",
    "@types/js-yaml": "4.0.9",
    "@types/normalize-path": "^3.0.2",
    "@types/read-pkg-up": "6.0.0",
    "@types/semver": "7.5.8",
    "@types/tunnel": "0.0.7",
    "@types/update-notifier": "6.0.8",
    "@types/url-parse": "^1.4.11",
    "@types/uuid": "10.0.0",
    "cross-env": "7.0.3",
    "deps": "^1.0.0",
    "typescript": "5.6.2",
    "vitest": "2.1.1"
  },
  "peerDependencies": {
    "@tsed/core": ">=7.14.2",
    "@tsed/di": ">=7.14.2"
  }
}
