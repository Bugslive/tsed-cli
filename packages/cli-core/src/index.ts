import Inquirer from "inquirer";
export * from "./CliCore.js";
export * from "./decorators/index.js";
export * from "./interfaces/index.js";
export * from "./packageManagers/index.js";
export * from "./services/index.js";
export * from "./utils/index.js";
export * from "@tsed/core";
export {
  Configuration,
  Constant,
  Container,
  DITest,
  Inject,
  Injectable,
  InjectorService,
  InvokeOptions,
  LocalsContainer,
  Module,
  OnDestroy,
  OnInit,
  Opts,
  OverrideProvider,
  registerProvider,
  TokenProvider,
  UseOpts,
  Value
} from "@tsed/di";
export * from "@tsed/logger";
export * from "@tsed/normalize-path";
export {Inquirer};

import "./utils/patchCommander";
