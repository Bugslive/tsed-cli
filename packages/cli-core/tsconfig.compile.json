{
  "extends": "../../tsconfig.compile.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "lib"
  },
  "exclude": [
    "node_modules",
    "test",
    "test/integrations/init/data",
    "lib",
    "**/*.spec.ts"
  ]
}
