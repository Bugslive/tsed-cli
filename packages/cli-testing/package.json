{
  "name": "@tsed/cli-testing",
  "description": "Utils to test you CLI based on Ts.ED CLI",
  "type": "commonjs",
  "version": "5.2.10",
  "source": "./src/index.ts",
  "main": "./lib/cjs/index.js",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "require": "./lib/cjs/index.js",
      "default": "./lib/esm/index.js"
    }
  },
  "scripts": {
    "build": "yarn build:ts",
    "build:ts": "tsc --build tsconfig.json",
    "test": "vitest run",
    "test:ci": "vitest run --coverage.thresholds.autoUpdate=true"
  },
  "keywords": [
    "Ts.ED",
    "cli",
    "typescript",
    "framework",
    "rest",
    "api",
    "express",
    "decorators"
  ],
  "dependencies": {
    "@tsed/cli-core": "5.2.10",
    "tslib": "2.3.1"
  },
  "devDependencies": {
    "@tsed/typescript": "workspace:*",
    "cross-env": "7.0.3",
    "typescript": "4.9.5",
    "vitest": "2.1.1"
  }
}
