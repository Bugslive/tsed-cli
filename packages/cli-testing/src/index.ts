export * from "./CliPlatformTest.js";
export * from "./FakeCliFs.js";
export * from "./FakeCliHttpClient.js";
export * from "./normalizePath.js";
