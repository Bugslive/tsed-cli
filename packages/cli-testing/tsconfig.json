{
  "extends": "@tsed/typescript/tsconfig.node.json",
  "compilerOptions": {
    "baseUrl": ".",
    "noEmit": true
  },
  "include": [],
  "references": [
    {
      "path": "../cli-core/tsconfig.json"
    },
    {
      "path": "./tsconfig.esm.json"
    }
  ]
}
