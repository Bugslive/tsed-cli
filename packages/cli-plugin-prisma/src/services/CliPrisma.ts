import {CliExeca, CliFs, Inject, ProjectPackageJson} from "@tsed/cli-core";
import {Injectable} from "@tsed/di";

@Injectable()
export class CliPrisma {
  @Inject()
  protected cliExeca: CliExeca;

  @Inject()
  protected cliFs: CliFs;

  @Inject()
  protected projectPackageJson: ProjectPackageJson;

  run(command: string, args: string[] = [], options: any = {}) {
    return this.cliExeca.run("npx", ["prisma", command, ...args], {
      ...options,
      cwd: this.projectPackageJson.dir
    });
  }

  async init() {
    return this.run("init");
  }

  async patchPrismaSchema() {
    const schemaPath = this.cliFs.join(this.projectPackageJson.dir, "prisma", "schema.prisma");

    if (this.cliFs.exists(schemaPath)) {
      let content = await this.cliFs.readFile(schemaPath, "utf8");

      if (!content.includes("generator tsed")) {
        content += "\ngenerator tsed {\n" + '  provider = "tsed-prisma"\n' + "}\n";

        return this.cliFs.writeFile(schemaPath, content, {encoding: "utf8"});
      }
    }
  }
}
