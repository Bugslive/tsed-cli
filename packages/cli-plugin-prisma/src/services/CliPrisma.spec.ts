import {CliPlatformTest} from "@tsed/cli-testing";
import {CliPrisma} from "./CliPrisma";
import {CliFs} from "@tsed/cli-core";
import {join} from "path";
import {normalizePath} from "@tsed/components-scan";

async function createServiceFixture() {
  const cliFs = {
    join: (...args: string[]) => normalizePath(join(...args)),
    exists: jest.fn(),
    readFile: jest.fn(),
    writeFile: jest.fn()
  };

  const service = await CliPlatformTest.invoke<CliPrisma>(CliPrisma, [
    {
      token: CliFs,
      use: cliFs
    }
  ]);

  return {service, cliFs};
}

describe("CliPrisma", () => {
  beforeEach(() =>
    CliPlatformTest.create({
      name: "tsed"
    })
  );
  afterEach(CliPlatformTest.reset);
  describe("patchPrismaSchema()", () => {
    it("should patch the schema", async () => {
      const {service, cliFs} = await createServiceFixture();
      const schema =
        "datasource db {\n" +
        '  provider = "sqlite"\n' +
        '  url      = env("DATABASE_URL")\n' +
        "}\n" +
        "\n" +
        "generator client {\n" +
        '  provider = "prisma-client-js"\n' +
        "}\n";

      cliFs.exists.mockReturnValue(true);
      cliFs.readFile.mockResolvedValue(schema);

      await service.patchPrismaSchema();

      expect(cliFs.writeFile).toHaveBeenCalledWith(
        expect.any(String),
        schema + "\ngenerator tsed {\n" + '  provider = "tsed-prisma"\n' + "}\n",
        {encoding: "utf8"}
      );
    });
    it("should not apply patch to the schema if it's already done", async () => {
      const {service, cliFs} = await createServiceFixture();
      const schema =
        "datasource db {\n" +
        '  provider = "sqlite"\n' +
        '  url      = env("DATABASE_URL")\n' +
        "}\n" +
        "\n" +
        "generator client {\n" +
        '  provider = "prisma-client-js"\n' +
        "}\n" +
        "\ngenerator tsed {\n" +
        '  provider = "tsed-prisma"\n' +
        "}\n";

      cliFs.exists.mockReturnValue(true);
      cliFs.readFile.mockResolvedValue(schema);

      await service.patchPrismaSchema();

      expect(cliFs.writeFile).not.toBeCalled();
    });
    it("should do nothing", async () => {
      const {service, cliFs} = await createServiceFixture();
      const schema =
        "datasource db {\n" +
        '  provider = "sqlite"\n' +
        '  url      = env("DATABASE_URL")\n' +
        "}\n" +
        "\n" +
        "generator client {\n" +
        '  provider = "prisma-client-js"\n' +
        "}\n";

      cliFs.exists.mockReturnValue(false);
      cliFs.readFile.mockResolvedValue(schema);

      await service.patchPrismaSchema();

      expect(cliFs.writeFile).not.toBeCalled();
    });
  });
});
