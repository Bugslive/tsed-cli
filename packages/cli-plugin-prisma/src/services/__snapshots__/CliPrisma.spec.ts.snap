// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CliPrisma > patchPrismaSchema() > should patch the schema 1`] = `
"datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

generator tsed {
  provider = "tsed-prisma"
}

model User {
  id    Int     @default(autoincrement()) @id
  email String  @unique
  name  String?
}
"
`;
